components/toast_wrapper/toast_wrapper.tsx

```diff
@@ -2,103 +2,95 @@
-        atBottom: PropTypes.bool,
-        lastViewedBottom: PropTypes.number,
-        width: PropTypes.number,
-        lastViewedAt: PropTypes.number,
-        focusedPostId: PropTypes.string,
-        initScrollOffsetFromBottom: PropTypes.number,
-        updateNewMessagesAtInChannel: PropTypes.func,
-        scrollToNewMessage: PropTypes.func,
-        scrollToLatestMessages: PropTypes.func,
-        scrollToUnreadMessages: PropTypes.func,
-        updateLastViewedBottomAt: PropTypes.func,
-        showSearchHintToast: PropTypes.bool,
-        onSearchHintDismiss: PropTypes.func,
-        shouldStartFromBottomWhenUnread: PropTypes.bool,
-        isNewMessageLineReached: PropTypes.bool,
-        unreadScrollPosition: PropTypes.string,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * Action creator to update toast status
-             */
-            updateToastStatus: PropTypes.func.isRequired,
-        }).isRequired,
-    };
-
+export type Props = PropsFromRedux & OwnProps & WrappedComponentProps & RouteComponentProps<{team: string}> & {
+    channelMarkedAsUnread?: boolean;
+    postListIds: string[];
+    latestPostTimeStamp?: number;
+    atBottom: boolean | null;
+    lastViewedBottom: number;
+    width: number;
+    focusedPostId?: string;
+    initScrollOffsetFromBottom: number;
+    updateNewMessagesAtInChannel: (lastViewedAt?: number) => void;
+    scrollToNewMessage: () => void;
+    scrollToLatestMessages: () => void;
+    scrollToUnreadMessages: () => void;
+    updateLastViewedBottomAt: (lastViewedBottom?: number) => void;
+    showSearchHintToast: boolean;
+    onSearchHintDismiss: () => void;
+    shouldStartFromBottomWhenUnread: boolean;
+    isNewMessageLineReached: boolean;
+};
+
+type State = {
+    unreadCount?: number;
+    unreadCountInChannel: number;
+    channelMarkedAsUnread?: boolean;
+    lastViewedAt?: number;
+    showUnreadToast?: boolean;
+    showNewMessagesToast?: boolean;
+    showMessageHistoryToast?: boolean;
+    showUnreadWithBottomStartToast?: boolean;
+};
+
+export class ToastWrapperClass extends React.PureComponent<Props, State> {
+    mounted?: boolean;
     static defaultProps = {
         focusedPostId: '',
     };
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
         this.state = {
             unreadCountInChannel: props.unreadCountInChannel,
         };
     }
 
-    static countNewMessages = (postListIds, rootPosts, isCollapsedThreadsEnabled) => {
-        const mark = getNewMessageIndex(postListIds);
+    static countNewMessages = (rootPosts: Record<string, boolean>, isCollapsedThreadsEnabled: boolean, postListIds?: string[]) => {
+        const mark = getNewMessageIndex(postListIds || []);
```

Comment:
nit: declare this above in the function arguments

Reply:
Done 👍